operations/fetch.go

```diff
@@ -466,6 +467,20 @@ func fileNameWithIndex(filename string, index int) string {
 	return fmt.Sprintf("%s_(%d).%s", parts[0], index-1, strings.Join(parts[1:], "."))
 }
 
+// truncateFilename truncates the filename (minus any extensions) so the entire filename length is less than the max
+func truncateFilename(fileName string) string {
+	if len(fileName) > fileNameMaxLength {
```

Comment:
I think there's an easier way to do this - use filepath.Ext to get the file extension, then strings.TrimSuffix to get the base name

Reply:
I think the question is whether we need to handle ".tar.gz". https://play.golang.org/p/hNOgmZR2xYM

Reply:
True, but we already break the filename up like this in fileNameWithIndex which wouldn't work for those filenames, no complaints yet :)